### PR TITLE
Submit single package with `phylum package`

### LINKED ISSUE #871: Submit single package
The API now supports [submitting a single package for analysis](https://api.staging.phylum.io/api/v0/swagger/index.html#/Packages/packages_submit_endpoint). This functionality should be made accessible from the CLI.

For example, we could add this directly to `phylum package` with a `--submit` flag.
```
phylum package --submit -t npm some-old-package 0.0.1
```

I welcome alternative suggestions for where we should add this functionality.

### PR DESCRIPTION
**Pull request by kylewillmon**
This patch modifies the `phylum package` command to use the [`/data/packages/submit`](https://api.phylum.io/api/v0/swagger/index.html#/Packages/packages_submit_endpoint) endpoint. That means that any requested package will be automatically submitted for processing if it is not already known to Phylum.

The new behavior looks like this:
```
# Brand new package
❯ phylum package -t npm left_pad 0.0.5
⚠️  Thank you for submitting this package. Please check back later for results.

# Package in processing
❯ phylum package -t npm left_pad 0.0.5
⚠️  Package is still processing. Please check back later for results.
```

Closes #871

### CODE REVIEW

**Comment by kylewillmon** [comment]
@louislang mentioned [in this comment](https://github.com/phylum-dev/cli/issues/871#issuecomment-1351790946) the he would like input from @furi0us333 on this issue. So we may want to wait for that feedback before merging this PR.

File: cli/src/api/mod.rs
Comment (cd-work) [inline cli/src/api/mod.rs:48]: This should probably be in `phylum_types`, right? Have you brought it up there?

File: cli/src/commands/packages.rs
Comment (cd-work) [inline cli/src/commands/packages.rs:22]:
```suggestion
        Some(package_type) => package_type.clone(),
```

File: cli/src/commands/extensions/api.rs
Comment (cd-work) [inline cli/src/commands/extensions/api.rs:342]: Why aren't we exposing all the enum variants to the user? Seems like useful information for an extension author.

**Review by cd-work — CHANGES_REQUESTED** [review]

File: cli/src/commands/extensions/api.rs
Comment (kylewillmon) [reply cli/src/commands/extensions/api.rs:342]: Maybe. But that's a breaking change to the extension API. Unless you just mean using different error messages for `New` vs `AlreadySubmitted`

**Review by maxrake — APPROVED** [review]
Functionality was confirmed with local testing. Nice new feature!

**Review by cd-work — APPROVED** [review]

**Pull request merged**
